<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <link href='https://fonts.googleapis.com/css?family=Chivo:900' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen" />
    <link rel="stylesheet" type="text/css" href="stylesheets/pygment_trac.css" media="screen" />
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print" />
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <title>PconsFold by ElofssonLab</title>
  </head>

  <body>
    <div id="container">
      <div class="inner">

        <header>
          <h1>PconsFold</h1>
          <h2>A pipeline for protein folding using predicted contacts from PconsC and Rosetta.</h2>
        </header>

        <section id="downloads" class="clearfix">
          <a href="https://github.com/ElofssonLab/pcons-fold/zipball/master" id="download-zip" class="button"><span>Download .zip</span></a>
          <a href="https://github.com/ElofssonLab/pcons-fold/tarball/master" id="download-tar-gz" class="button"><span>Download .tar.gz</span></a>
          <a href="https://github.com/ElofssonLab/pcons-fold" id="view-on-github" class="button"><span>View on GitHub</span></a>
        </section>

        <hr>

        <section id="main_content">
          <h2>
<a name="pipeline-overview" class="anchor" href="#pipeline-overview"><span class="octicon octicon-link"></span></a>Pipeline overview:</h2>

<ol>
<li>
<strong>Input:</strong> fasta file containing one protein sequence</li>
<li>Prepare input for PconsC</li>
<li>Contact prediction with PconsC</li>
<li>Prepare input for Rosetta folding</li>
<li>Rosetta folding</li>
<li>Extract and relax structures with lowest Rosetta energy</li>
<li>
<strong>Output:</strong> the predicted contact map (also as a plot) and the top-ranked structural model(s) relaxed and non-relaxed</li>
</ol><h2>
<a name="dependencies" class="anchor" href="#dependencies"><span class="octicon octicon-link"></span></a>Dependencies:</h2>

<ul>
<li><a href="https://www.rosettacommons.org/software">Rosetta v3.5 or weekly built</a></li>
<li><a href="http://hmmer.janelia.org/">Jackhmmer from HMMER v3.0 or higher</a></li>
<li><a href="http://toolkit.tuebingen.mpg.de/hhblits">HHblits from HHsuite v2.0.16</a></li>
<li><a href="http://bioinfadmin.cs.ucl.ac.uk/downloads/PSICOV/">PSICOV v1.11</a></li>
<li><a href="http://plmdca.csc.kth.se/">plmDCA symmetric v2</a></li>
<li>either MATLAB v8.1 or higher</li>
<li>or <a href="http://www.mathworks.se/products/compiler/mcr/">MATLAB Compiler Runtime (MCR) v8.1</a>
</li>
</ul><p>MATLAB is needed to run plmDCA. However, if MATLAB is not available you can also use a compiled version of plmDCA. For the compiled version to run you need to provide a path to MCR.</p>

<h2>
<a name="how-to-run-it" class="anchor" href="#how-to-run-it"><span class="octicon octicon-link"></span></a>How to run it:</h2>

<p><strong>Make sure</strong> all dependencies are working correctly and adjust the paths in <code>localconfig.py</code>.</p>

<p>To run the full pipeline use:</p>

<pre><code>./pcons_fold.py [-c n_cores] [-n n_decoys] [-m n_models]
                [-f factor] [--norelax] [--nohoms] 
                hhblits_database jackhmmer_database sequence_file
</code></pre>

<ul>
<li>Required:

<ul>
<li>
<code>hhblits_database</code> and <code>jackhmmer_database</code> are paths to the databases used by HHblits and Jackhmmer</li>
<li>
<code>sequence_file</code> is the path to the input protein sequence in FASTA format (only single sequences). </li>
</ul>
</li>
<li>Optional:

<ul>
<li>
<code>n_cores</code> specifies the number of cores to use during computation (default: number of available cores). </li>
<li>
<code>n_decoys</code> specifies  the number of decoy structures generated by Rosetta (default: 2000, see publication). </li>
<li>
<code>n_models</code> is the number of top-ranked models being extracted and eventually relaxed in the end (default: 10).</li>
<li>
<code>factor</code> determines the number of constraints used to fold the protein, which is: <code>factor</code> * length_of_the_input_sequence (default: 1.0). </li>
<li>
<code>norelax</code> is a flag that supresses relaxation of the final models. This can be used to quickly extract structures in the end. </li>
<li>
<code>nohoms</code> is a flag that ensures that homologous structures are excluded from fragment picking. This is only useful in test cases if the model quality needs to be evaluated with a known structure.</li>
</ul>
</li>
</ul><hr><p>You can also <strong>run PconsC contact prediction</strong> independently with this command:</p>

<pre><code>./pconsc/predict_all.py [-c cores] hhblits_database jackhmmer_database sequence_file
</code></pre>

<p>And then <strong>fold the protein</strong> according to given predicted contacts with the following commands:</p>

<pre><code>./folding/rosetta/prepare_input.py [-f factor] [--nohoms] sequence_file contact_map 

./folding/rosetta/fold.py [-c n_cores] [-n n_decoys] sequence_file rosetta_constraintfile

./folding/rosetta/extract.py [-c n_cores] [-m n_models] [--norelax] number_of_extracted_structures
</code></pre>

<p>The first script generates the file <code>(pconsc_output)-(factor).constraints</code> which is then used by Rosetta in the next step with <code>rosetta_constraintfile</code>.</p>
        </section>

        <footer>
          PconsFold is maintained by <a href="https://github.com/ElofssonLab">ElofssonLab</a><br>
          This page was generated by <a href="http://pages.github.com">GitHub Pages</a>. Tactile theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.
        </footer>

        
      </div>
    </div>
  </body>
</html>