PconsFold
=========

A pipeline for protein folding using predicted contacts from PconsC and a  Rosetta folding protocol.

![PconsFold pipeline](https://github.com/ElofssonLab/pcons-fold/blob/release-1.0/pipeline_horiz.png)


## Pipeline overview:

1. Input: fasta file containing one sequence
2. Prepare input for PconsC
3. Contact prediction with PconsC
4. Prepare input for Rosetta folding
5. Rosetta folding
6. Extract and relax structures with lowest Rosetta energy
7. Output: the predicted contact map (also as a plot) and the lowest energy structure.


## Dependencies:

- Rosetta v3.5 or weekly
- Jackhmmer from HMMER v3.0 or higher
- HHblits from HHsuite v2.0.16 or higher
- PSICOV v1.11 or higher
- either MATLAB v8.1 or higher
- or MATLAB Compiler Runtime (MCR) v8.1 or higher 

MATLAB is needed to run plmDCA. However, if MATLAB is not available you can also use a compiled version of plmDCA. For the compiled version to run you need to provide a path to MCR.


## How to run it:

__Make sure__ all paths are correct in `localconfig.py`. Remark: except for plmDCA, dependencies are not included in this repository.

To run the full pipeline use:
```
./pcons_fold.py [-c n_cores] [-n n_decoys] [-m n_models]
                  [-f factor] [--norelax] [--nohoms] 
                  hhblits_database jackhmmer_database sequence_file
```
- Required:
  - `hhblits_database` and `jackhmmer_database` are paths to the databases used by HHblits and Jackhmmer
  - `sequence_file` is the path to the input protein sequence in FASTA format (only single sequences). 
- Optional:
  - `n_cores` specifies the number of cores to use during computation (default: number of available cores). 
  - `n_decoys` specifies  the number of decoy structures generated by Rosetta (default: 2000, see publication). 
  - `n_models` is the number of top-ranked models being extracted and eventually relaxed in the end (default: 10).
  - `factor` determines the number of constraints used to fold the protein, which is: `factor` * `length of the input sequence` (default: 1.0). 
  - `norelax` is a flag that supresses relaxation of the final models. This can be used to quickly extract structures in the end. 
  - `nohoms` is a flag that ensures that homologous structures are excluded from fragment picking. This is only useful in test cases if the model quality needs to be evaluated with a known structure.


---


You can also run contact prediction independently with this command:
```
./pconsc/predict_all.py [-c cores] hhblits_database jackhmmer_database sequence_file
```


And then fold the protein according to given predicted contacts with the following commands:
``` 
./folding/rosetta/prepare_input.py [-f factor] [--nohoms] sequence_file contact_map 

./folding/rosetta/fold.py [-c n_cores] [-n n_decoys] sequence_file rosetta_constraintfile

./folding/rosetta/extract.py [-c n_cores] [-m n_models] [--norelax] number_of_extracted_structures
```

The first script generates the file `(pconsc_output)-(factor).constraints` which is then used by Rosetta in the next step with `rosetta_constraintfile`.


